avm/res/event-hub/namespace/eventhub/main.bicep

```diff
@@ -0,0 +1,265 @@
+  name: name
+  parent: namespace
+  properties: captureDescriptionEnabled ? union(eventHubProperties, eventHubPropertiesCapture) : eventHubProperties
+}
+
+resource eventHub_lock 'Microsoft.Authorization/locks@2020-05-01' = if (!empty(lock ?? {}) && lock.?kind != 'None') {
+  name: lock.?name ?? 'lock-${name}'
+  properties: {
+    level: lock.?kind ?? ''
+    notes: lock.?kind == 'CanNotDelete' ? 'Cannot delete resource or child resources.' : 'Cannot delete or modify the resource or child resources.'
+  }
+  scope: eventHub
+}
+
+module eventHub_consumergroups 'consumergroup/main.bicep' = [for (consumerGroup, index) in consumergroups: {
+  name: '${deployment().name}-ConsumerGroup-${index}'
+  params: {
+    namespaceName: namespaceName
+    eventHubName: eventHub.name
+    name: consumerGroup.name
+    userMetadata: contains(consumerGroup, 'userMetadata') ? consumerGroup.userMetadata : ''
+  }
+}]
+
+module eventHub_authorizationRules 'authorization-rule/main.bicep' = [for (authorizationRule, index) in authorizationRules: {
+  name: '${deployment().name}-AuthRule-${index}'
+  params: {
+    namespaceName: namespaceName
+    eventHubName: eventHub.name
+    name: authorizationRule.name
+    rights: contains(authorizationRule, 'rights') ? authorizationRule.rights : []
+  }
+}]
+
+resource eventHub_roleAssignments 'Microsoft.Authorization/roleAssignments@2022-04-01' = [for (roleAssignment, index) in (roleAssignments ?? []): {
+  name: guid(eventHub.id, roleAssignment.principalId, roleAssignment.roleDefinitionIdOrName)
+  properties: {
+    roleDefinitionId: contains(builtInRoleNames, roleAssignment.roleDefinitionIdOrName) ? builtInRoleNames[roleAssignment.roleDefinitionIdOrName] : contains(roleAssignment.roleDefinitionIdOrName, '/providers/Microsoft.Authorization/roleDefinitions/') ? roleAssignment.roleDefinitionIdOrName : subscriptionResourceId('Microsoft.Authorization/roleDefinitions', roleAssignment.roleDefinitionIdOrName)
+    principalId: roleAssignment.principalId
+    description: roleAssignment.?description
+    principalType: roleAssignment.?principalType
+    condition: roleAssignment.?condition
+    conditionVersion: !empty(roleAssignment.?condition) ? (roleAssignment.?conditionVersion ?? '2.0') : null // Must only be set if condtion is set
+    delegatedManagedIdentityResourceId: roleAssignment.?delegatedManagedIdentityResourceId
+  }
+  scope: eventHub
+}]
+
+@description('The name of the event hub.')
+output name string = eventHub.name
+
+@description('The resource ID of the event hub.')
+output eventHubId string = eventHub.id
+
+@description('The resource group the event hub was deployed into.')
+output resourceGroupName string = resourceGroup().name
+
+@description('The authentication rule resource ID of the event hub.')
+output resourceId string = az.resourceId('Microsoft.EventHub/namespaces/authorizationRules', namespaceName, 'RootManageSharedAccessKey')
```

Comment:
```suggestion
output authorizationRuleResourceId string = az.resourceId('Microsoft.EventHub/namespaces/authorizationRules', namespaceName, 'RootManageSharedAccessKey')
```

Reply:
I wonder why this output exists though. If the default parameter is overwritten, this WILL fail. I would suggest to remove it

Reply:
updated